// vite-advance-api@1.0.18/dist/index.es.js
import k from "express";
import zt from "cors";
import Vt from "axios";
import { v4 as qt } from "uuid";
import "express-async-errors";
var z = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function pe(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var Yt = Array.isArray, S = Yt, Wt = typeof z == "object" && z && z.Object === Object && z, yt = Wt, Jt = yt, Xt = typeof self == "object" && self && self.Object === Object && self, Zt = Jt || Xt || Function("return this")(), $ = Zt, Qt = $, er = Qt.Symbol, U = er, Ie = U, $t = Object.prototype, tr = $t.hasOwnProperty, rr = $t.toString, D = Ie ? Ie.toStringTag : void 0;
function ar(e) {
  var t = tr.call(e, D), r = e[D];
  try {
    e[D] = void 0;
    var a = !0;
  } catch {
  }
  var n = rr.call(e);
  return a && (t ? e[D] = r : delete e[D]), n;
}
var nr = ar, or = Object.prototype, sr = or.toString;
function ir(e) {
  return sr.call(e);
}
var cr = ir, Ee = U, lr = nr, ur = cr, fr = "[object Null]", pr = "[object Undefined]", Me = Ee ? Ee.toStringTag : void 0;
function vr(e) {
  return e == null ? e === void 0 ? pr : fr : Me && Me in Object(e) ? lr(e) : ur(e);
}
var I = vr;
function gr(e) {
  return e != null && typeof e == "object";
}
var w = gr, dr = I, hr = w, br = "[object Symbol]";
function yr(e) {
  return typeof e == "symbol" || hr(e) && dr(e) == br;
}
var ve = yr, $r = S, mr = ve, _r = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, Tr = /^\w*$/;
function xr(e, t) {
  if ($r(e))
    return !1;
  var r = typeof e;
  return r == "number" || r == "symbol" || r == "boolean" || e == null || mr(e) ? !0 : Tr.test(e) || !_r.test(e) || t != null && e in Object(t);
}
var Ar = xr;
function Sr(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
var E = Sr, Or = I, wr = E, Cr = "[object AsyncFunction]", jr = "[object Function]", Pr = "[object GeneratorFunction]", Ir = "[object Proxy]";
function Er(e) {
  if (!wr(e))
    return !1;
  var t = Or(e);
  return t == jr || t == Pr || t == Cr || t == Ir;
}
var mt = Er, Mr = $, Br = Mr["__core-js_shared__"], Fr = Br, re = Fr, Be = function() {
  var e = /[^.]+$/.exec(re && re.keys && re.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function Lr(e) {
  return !!Be && Be in e;
}
var Rr = Lr, Gr = Function.prototype, kr = Gr.toString;
function Dr(e) {
  if (e != null) {
    try {
      return kr.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var _t = Dr, Nr = mt, Ur = Rr, Kr = E, Hr = _t, zr = /[\\^$.*+?()[\]{}|]/g, Vr = /^\[object .+?Constructor\]$/, qr = Function.prototype, Yr = Object.prototype, Wr = qr.toString, Jr = Yr.hasOwnProperty, Xr = RegExp(
  "^" + Wr.call(Jr).replace(zr, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function Zr(e) {
  if (!Kr(e) || Ur(e))
    return !1;
  var t = Nr(e) ? Xr : Vr;
  return t.test(Hr(e));
}
var Qr = Zr;
function ea(e, t) {
  return e == null ? void 0 : e[t];
}
var ta = ea, ra = Qr, aa = ta;
function na(e, t) {
  var r = aa(e, t);
  return ra(r) ? r : void 0;
}
var C = na, oa = C, sa = oa(Object, "create"), X = sa, Fe = X;
function ia() {
  this.__data__ = Fe ? Fe(null) : {}, this.size = 0;
}
var ca = ia;
function la(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var ua = la, fa = X, pa = "__lodash_hash_undefined__", va = Object.prototype, ga = va.hasOwnProperty;
function da(e) {
  var t = this.__data__;
  if (fa) {
    var r = t[e];
    return r === pa ? void 0 : r;
  }
  return ga.call(t, e) ? t[e] : void 0;
}
var ha = da, ba = X, ya = Object.prototype, $a = ya.hasOwnProperty;
function ma(e) {
  var t = this.__data__;
  return ba ? t[e] !== void 0 : $a.call(t, e);
}
var _a = ma, Ta = X, xa = "__lodash_hash_undefined__";
function Aa(e, t) {
  var r = this.__data__;
  return this.size += this.has(e) ? 0 : 1, r[e] = Ta && t === void 0 ? xa : t, this;
}
var Sa = Aa, Oa = ca, wa = ua, Ca = ha, ja = _a, Pa = Sa;
function M(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var a = e[t];
    this.set(a[0], a[1]);
  }
}
M.prototype.clear = Oa;
M.prototype.delete = wa;
M.prototype.get = Ca;
M.prototype.has = ja;
M.prototype.set = Pa;
var Ia = M;
function Ea() {
  this.__data__ = [], this.size = 0;
}
var Ma = Ea;
function Ba(e, t) {
  return e === t || e !== e && t !== t;
}
var Tt = Ba, Fa = Tt;
function La(e, t) {
  for (var r = e.length; r--; )
    if (Fa(e[r][0], t))
      return r;
  return -1;
}
var Z = La, Ra = Z, Ga = Array.prototype, ka = Ga.splice;
function Da(e) {
  var t = this.__data__, r = Ra(t, e);
  if (r < 0)
    return !1;
  var a = t.length - 1;
  return r == a ? t.pop() : ka.call(t, r, 1), --this.size, !0;
}
var Na = Da, Ua = Z;
function Ka(e) {
  var t = this.__data__, r = Ua(t, e);
  return r < 0 ? void 0 : t[r][1];
}
var Ha = Ka, za = Z;
function Va(e) {
  return za(this.__data__, e) > -1;
}
var qa = Va, Ya = Z;
function Wa(e, t) {
  var r = this.__data__, a = Ya(r, e);
  return a < 0 ? (++this.size, r.push([e, t])) : r[a][1] = t, this;
}
var Ja = Wa, Xa = Ma, Za = Na, Qa = Ha, en = qa, tn = Ja;
function B(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var a = e[t];
    this.set(a[0], a[1]);
  }
}
B.prototype.clear = Xa;
B.prototype.delete = Za;
B.prototype.get = Qa;
B.prototype.has = en;
B.prototype.set = tn;
var Q = B, rn = C, an = $, nn = rn(an, "Map"), ge = nn, Le = Ia, on = Q, sn = ge;
function cn() {
  this.size = 0, this.__data__ = {
    hash: new Le(),
    map: new (sn || on)(),
    string: new Le()
  };
}
var ln = cn;
function un(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
var fn = un, pn = fn;
function vn(e, t) {
  var r = e.__data__;
  return pn(t) ? r[typeof t == "string" ? "string" : "hash"] : r.map;
}
var ee = vn, gn = ee;
function dn(e) {
  var t = gn(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
var hn = dn, bn = ee;
function yn(e) {
  return bn(this, e).get(e);
}
var $n = yn, mn = ee;
function _n(e) {
  return mn(this, e).has(e);
}
var Tn = _n, xn = ee;
function An(e, t) {
  var r = xn(this, e), a = r.size;
  return r.set(e, t), this.size += r.size == a ? 0 : 1, this;
}
var Sn = An, On = ln, wn = hn, Cn = $n, jn = Tn, Pn = Sn;
function F(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var a = e[t];
    this.set(a[0], a[1]);
  }
}
F.prototype.clear = On;
F.prototype.delete = wn;
F.prototype.get = Cn;
F.prototype.has = jn;
F.prototype.set = Pn;
var xt = F, At = xt, In = "Expected a function";
function de(e, t) {
  if (typeof e != "function" || t != null && typeof t != "function")
    throw new TypeError(In);
  var r = function() {
    var a = arguments, n = t ? t.apply(this, a) : a[0], s = r.cache;
    if (s.has(n))
      return s.get(n);
    var o = e.apply(this, a);
    return r.cache = s.set(n, o) || s, o;
  };
  return r.cache = new (de.Cache || At)(), r;
}
de.Cache = At;
var En = de, Mn = En, Bn = 500;
function Fn(e) {
  var t = Mn(e, function(a) {
    return r.size === Bn && r.clear(), a;
  }), r = t.cache;
  return t;
}
var Ln = Fn, Rn = Ln, Gn = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, kn = /\\(\\)?/g, Dn = Rn(function(e) {
  var t = [];
  return e.charCodeAt(0) === 46 && t.push(""), e.replace(Gn, function(r, a, n, s) {
    t.push(n ? s.replace(kn, "$1") : a || r);
  }), t;
}), Nn = Dn;
function Un(e, t) {
  for (var r = -1, a = e == null ? 0 : e.length, n = Array(a); ++r < a; )
    n[r] = t(e[r], r, e);
  return n;
}
var St = Un, Re = U, Kn = St, Hn = S, zn = ve, Ge = Re ? Re.prototype : void 0, ke = Ge ? Ge.toString : void 0;
function Ot(e) {
  if (typeof e == "string")
    return e;
  if (Hn(e))
    return Kn(e, Ot) + "";
  if (zn(e))
    return ke ? ke.call(e) : "";
  var t = e + "";
  return t == "0" && 1 / e == -1 / 0 ? "-0" : t;
}
var Vn = Ot, qn = Vn;
function Yn(e) {
  return e == null ? "" : qn(e);
}
var Wn = Yn, Jn = S, Xn = Ar, Zn = Nn, Qn = Wn;
function eo(e, t) {
  return Jn(e) ? e : Xn(e, t) ? [e] : Zn(Qn(e));
}
var L = eo, to = ve;
function ro(e) {
  if (typeof e == "string" || to(e))
    return e;
  var t = e + "";
  return t == "0" && 1 / e == -1 / 0 ? "-0" : t;
}
var te = ro, ao = L, no = te;
function oo(e, t) {
  t = ao(t, e);
  for (var r = 0, a = t.length; e != null && r < a; )
    e = e[no(t[r++])];
  return r && r == a ? e : void 0;
}
var he = oo, so = C, io = function() {
  try {
    var e = so(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}(), wt = io, De = wt;
function co(e, t, r) {
  t == "__proto__" && De ? De(e, t, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : e[t] = r;
}
var Ct = co, lo = Ct, uo = Tt, fo = Object.prototype, po = fo.hasOwnProperty;
function vo(e, t, r) {
  var a = e[t];
  (!(po.call(e, t) && uo(a, r)) || r === void 0 && !(t in e)) && lo(e, t, r);
}
var be = vo, go = 9007199254740991, ho = /^(?:0|[1-9]\d*)$/;
function bo(e, t) {
  var r = typeof e;
  return t = t ?? go, !!t && (r == "number" || r != "symbol" && ho.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
var ye = bo, yo = be, $o = L, mo = ye, Ne = E, _o = te;
function To(e, t, r, a) {
  if (!Ne(e))
    return e;
  t = $o(t, e);
  for (var n = -1, s = t.length, o = s - 1, i = e; i != null && ++n < s; ) {
    var c = _o(t[n]), l = r;
    if (c === "__proto__" || c === "constructor" || c === "prototype")
      return e;
    if (n != o) {
      var u = i[c];
      l = a ? a(u, c, i) : void 0, l === void 0 && (l = Ne(u) ? u : mo(t[n + 1]) ? [] : {});
    }
    yo(i, c, l), i = i[c];
  }
  return e;
}
var xo = To, Ao = he, So = xo, Oo = L;
function wo(e, t, r) {
  for (var a = -1, n = t.length, s = {}; ++a < n; ) {
    var o = t[a], i = Ao(e, o);
    r(i, o) && So(s, Oo(o, e), i);
  }
  return s;
}
var Co = wo;
function jo(e, t) {
  return e != null && t in Object(e);
}
var Po = jo, Io = I, Eo = w, Mo = "[object Arguments]";
function Bo(e) {
  return Eo(e) && Io(e) == Mo;
}
var Fo = Bo, Ue = Fo, Lo = w, jt = Object.prototype, Ro = jt.hasOwnProperty, Go = jt.propertyIsEnumerable, ko = Ue(/* @__PURE__ */ function() {
  return arguments;
}()) ? Ue : function(e) {
  return Lo(e) && Ro.call(e, "callee") && !Go.call(e, "callee");
}, $e = ko, Do = 9007199254740991;
function No(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Do;
}
var me = No, Uo = L, Ko = $e, Ho = S, zo = ye, Vo = me, qo = te;
function Yo(e, t, r) {
  t = Uo(t, e);
  for (var a = -1, n = t.length, s = !1; ++a < n; ) {
    var o = qo(t[a]);
    if (!(s = e != null && r(e, o)))
      break;
    e = e[o];
  }
  return s || ++a != n ? s : (n = e == null ? 0 : e.length, !!n && Vo(n) && zo(o, n) && (Ho(e) || Ko(e)));
}
var Wo = Yo, Jo = Po, Xo = Wo;
function Zo(e, t) {
  return e != null && Xo(e, t, Jo);
}
var Qo = Zo, es = Co, ts = Qo;
function rs(e, t) {
  return es(e, t, function(r, a) {
    return ts(e, a);
  });
}
var as = rs;
function ns(e, t) {
  for (var r = -1, a = t.length, n = e.length; ++r < a; )
    e[n + r] = t[r];
  return e;
}
var _e = ns, Ke = U, os = $e, ss = S, He = Ke ? Ke.isConcatSpreadable : void 0;
function is(e) {
  return ss(e) || os(e) || !!(He && e && e[He]);
}
var cs = is, ls = _e, us = cs;
function Pt(e, t, r, a, n) {
  var s = -1, o = e.length;
  for (r || (r = us), n || (n = []); ++s < o; ) {
    var i = e[s];
    t > 0 && r(i) ? t > 1 ? Pt(i, t - 1, r, a, n) : ls(n, i) : a || (n[n.length] = i);
  }
  return n;
}
var fs = Pt, ps = fs;
function vs(e) {
  var t = e == null ? 0 : e.length;
  return t ? ps(e, 1) : [];
}
var gs = vs;
function ds(e, t, r) {
  switch (r.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, r[0]);
    case 2:
      return e.call(t, r[0], r[1]);
    case 3:
      return e.call(t, r[0], r[1], r[2]);
  }
  return e.apply(t, r);
}
var hs = ds, bs = hs, ze = Math.max;
function ys(e, t, r) {
  return t = ze(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var a = arguments, n = -1, s = ze(a.length - t, 0), o = Array(s); ++n < s; )
      o[n] = a[t + n];
    n = -1;
    for (var i = Array(t + 1); ++n < t; )
      i[n] = a[n];
    return i[t] = r(o), bs(e, this, i);
  };
}
var $s = ys;
function ms(e) {
  return function() {
    return e;
  };
}
var _s = ms;
function Ts(e) {
  return e;
}
var xs = Ts, As = _s, Ve = wt, Ss = xs, Os = Ve ? function(e, t) {
  return Ve(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: As(t),
    writable: !0
  });
} : Ss, ws = Os, Cs = 800, js = 16, Ps = Date.now;
function Is(e) {
  var t = 0, r = 0;
  return function() {
    var a = Ps(), n = js - (a - r);
    if (r = a, n > 0) {
      if (++t >= Cs)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
var Es = Is, Ms = ws, Bs = Es, Fs = Bs(Ms), Ls = Fs, Rs = gs, Gs = $s, ks = Ls;
function Ds(e) {
  return ks(Gs(e, void 0, Rs), e + "");
}
var It = Ds, Ns = as, Us = It, Ks = Us(function(e, t) {
  return e == null ? {} : Ns(e, t);
}), Hs = Ks;
const zs = /* @__PURE__ */ pe(Hs);
var Vs = Q;
function qs() {
  this.__data__ = new Vs(), this.size = 0;
}
var Ys = qs;
function Ws(e) {
  var t = this.__data__, r = t.delete(e);
  return this.size = t.size, r;
}
var Js = Ws;
function Xs(e) {
  return this.__data__.get(e);
}
var Zs = Xs;
function Qs(e) {
  return this.__data__.has(e);
}
var ei = Qs, ti = Q, ri = ge, ai = xt, ni = 200;
function oi(e, t) {
  var r = this.__data__;
  if (r instanceof ti) {
    var a = r.__data__;
    if (!ri || a.length < ni - 1)
      return a.push([e, t]), this.size = ++r.size, this;
    r = this.__data__ = new ai(a);
  }
  return r.set(e, t), this.size = r.size, this;
}
var si = oi, ii = Q, ci = Ys, li = Js, ui = Zs, fi = ei, pi = si;
function R(e) {
  var t = this.__data__ = new ii(e);
  this.size = t.size;
}
R.prototype.clear = ci;
R.prototype.delete = li;
R.prototype.get = ui;
R.prototype.has = fi;
R.prototype.set = pi;
var vi = R;
function gi(e, t) {
  for (var r = -1, a = e == null ? 0 : e.length; ++r < a && t(e[r], r, e) !== !1; )
    ;
  return e;
}
var di = gi, hi = be, bi = Ct;
function yi(e, t, r, a) {
  var n = !r;
  r || (r = {});
  for (var s = -1, o = t.length; ++s < o; ) {
    var i = t[s], c = a ? a(r[i], e[i], i, r, e) : void 0;
    c === void 0 && (c = e[i]), n ? bi(r, i, c) : hi(r, i, c);
  }
  return r;
}
var K = yi;
function $i(e, t) {
  for (var r = -1, a = Array(e); ++r < e; )
    a[r] = t(r);
  return a;
}
var mi = $i, q = { exports: {} };
function _i() {
  return !1;
}
var Ti = _i;
q.exports;
(function(e, t) {
  var r = $, a = Ti, n = t && !t.nodeType && t, s = n && !0 && e && !e.nodeType && e, o = s && s.exports === n, i = o ? r.Buffer : void 0, c = i ? i.isBuffer : void 0, l = c || a;
  e.exports = l;
})(q, q.exports);
var Et = q.exports, xi = I, Ai = me, Si = w, Oi = "[object Arguments]", wi = "[object Array]", Ci = "[object Boolean]", ji = "[object Date]", Pi = "[object Error]", Ii = "[object Function]", Ei = "[object Map]", Mi = "[object Number]", Bi = "[object Object]", Fi = "[object RegExp]", Li = "[object Set]", Ri = "[object String]", Gi = "[object WeakMap]", ki = "[object ArrayBuffer]", Di = "[object DataView]", Ni = "[object Float32Array]", Ui = "[object Float64Array]", Ki = "[object Int8Array]", Hi = "[object Int16Array]", zi = "[object Int32Array]", Vi = "[object Uint8Array]", qi = "[object Uint8ClampedArray]", Yi = "[object Uint16Array]", Wi = "[object Uint32Array]", p = {};
p[Ni] = p[Ui] = p[Ki] = p[Hi] = p[zi] = p[Vi] = p[qi] = p[Yi] = p[Wi] = !0;
p[Oi] = p[wi] = p[ki] = p[Ci] = p[Di] = p[ji] = p[Pi] = p[Ii] = p[Ei] = p[Mi] = p[Bi] = p[Fi] = p[Li] = p[Ri] = p[Gi] = !1;
function Ji(e) {
  return Si(e) && Ai(e.length) && !!p[xi(e)];
}
var Xi = Ji;
function Zi(e) {
  return function(t) {
    return e(t);
  };
}
var Te = Zi, Y = { exports: {} };
Y.exports;
(function(e, t) {
  var r = yt, a = t && !t.nodeType && t, n = a && !0 && e && !e.nodeType && e, s = n && n.exports === a, o = s && r.process, i = function() {
    try {
      var c = n && n.require && n.require("util").types;
      return c || o && o.binding && o.binding("util");
    } catch {
    }
  }();
  e.exports = i;
})(Y, Y.exports);
var xe = Y.exports, Qi = Xi, ec = Te, qe = xe, Ye = qe && qe.isTypedArray, tc = Ye ? ec(Ye) : Qi, rc = tc, ac = mi, nc = $e, oc = S, sc = Et, ic = ye, cc = rc, lc = Object.prototype, uc = lc.hasOwnProperty;
function fc(e, t) {
  var r = oc(e), a = !r && nc(e), n = !r && !a && sc(e), s = !r && !a && !n && cc(e), o = r || a || n || s, i = o ? ac(e.length, String) : [], c = i.length;
  for (var l in e)
    (t || uc.call(e, l)) && !(o && // Safari 9 has enumerable `arguments.length` in strict mode.
    (l == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    n && (l == "offset" || l == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    s && (l == "buffer" || l == "byteLength" || l == "byteOffset") || // Skip index properties.
    ic(l, c))) && i.push(l);
  return i;
}
var Mt = fc, pc = Object.prototype;
function vc(e) {
  var t = e && e.constructor, r = typeof t == "function" && t.prototype || pc;
  return e === r;
}
var Ae = vc;
function gc(e, t) {
  return function(r) {
    return e(t(r));
  };
}
var Bt = gc, dc = Bt, hc = dc(Object.keys, Object), bc = hc, yc = Ae, $c = bc, mc = Object.prototype, _c = mc.hasOwnProperty;
function Tc(e) {
  if (!yc(e))
    return $c(e);
  var t = [];
  for (var r in Object(e))
    _c.call(e, r) && r != "constructor" && t.push(r);
  return t;
}
var xc = Tc, Ac = mt, Sc = me;
function Oc(e) {
  return e != null && Sc(e.length) && !Ac(e);
}
var Ft = Oc, wc = Mt, Cc = xc, jc = Ft;
function Pc(e) {
  return jc(e) ? wc(e) : Cc(e);
}
var Se = Pc, Ic = K, Ec = Se;
function Mc(e, t) {
  return e && Ic(t, Ec(t), e);
}
var Bc = Mc;
function Fc(e) {
  var t = [];
  if (e != null)
    for (var r in Object(e))
      t.push(r);
  return t;
}
var Lc = Fc, Rc = E, Gc = Ae, kc = Lc, Dc = Object.prototype, Nc = Dc.hasOwnProperty;
function Uc(e) {
  if (!Rc(e))
    return kc(e);
  var t = Gc(e), r = [];
  for (var a in e)
    a == "constructor" && (t || !Nc.call(e, a)) || r.push(a);
  return r;
}
var Kc = Uc, Hc = Mt, zc = Kc, Vc = Ft;
function qc(e) {
  return Vc(e) ? Hc(e, !0) : zc(e);
}
var Oe = qc, Yc = K, Wc = Oe;
function Jc(e, t) {
  return e && Yc(t, Wc(t), e);
}
var Xc = Jc, W = { exports: {} };
W.exports;
(function(e, t) {
  var r = $, a = t && !t.nodeType && t, n = a && !0 && e && !e.nodeType && e, s = n && n.exports === a, o = s ? r.Buffer : void 0, i = o ? o.allocUnsafe : void 0;
  function c(l, u) {
    if (u)
      return l.slice();
    var g = l.length, d = i ? i(g) : new l.constructor(g);
    return l.copy(d), d;
  }
  e.exports = c;
})(W, W.exports);
var Zc = W.exports;
function Qc(e, t) {
  var r = -1, a = e.length;
  for (t || (t = Array(a)); ++r < a; )
    t[r] = e[r];
  return t;
}
var el = Qc;
function tl(e, t) {
  for (var r = -1, a = e == null ? 0 : e.length, n = 0, s = []; ++r < a; ) {
    var o = e[r];
    t(o, r, e) && (s[n++] = o);
  }
  return s;
}
var rl = tl;
function al() {
  return [];
}
var Lt = al, nl = rl, ol = Lt, sl = Object.prototype, il = sl.propertyIsEnumerable, We = Object.getOwnPropertySymbols, cl = We ? function(e) {
  return e == null ? [] : (e = Object(e), nl(We(e), function(t) {
    return il.call(e, t);
  }));
} : ol, we = cl, ll = K, ul = we;
function fl(e, t) {
  return ll(e, ul(e), t);
}
var pl = fl, vl = Bt, gl = vl(Object.getPrototypeOf, Object), Ce = gl, dl = _e, hl = Ce, bl = we, yl = Lt, $l = Object.getOwnPropertySymbols, ml = $l ? function(e) {
  for (var t = []; e; )
    dl(t, bl(e)), e = hl(e);
  return t;
} : yl, Rt = ml, _l = K, Tl = Rt;
function xl(e, t) {
  return _l(e, Tl(e), t);
}
var Al = xl, Sl = _e, Ol = S;
function wl(e, t, r) {
  var a = t(e);
  return Ol(e) ? a : Sl(a, r(e));
}
var Gt = wl, Cl = Gt, jl = we, Pl = Se;
function Il(e) {
  return Cl(e, Pl, jl);
}
var El = Il, Ml = Gt, Bl = Rt, Fl = Oe;
function Ll(e) {
  return Ml(e, Fl, Bl);
}
var kt = Ll, Rl = C, Gl = $, kl = Rl(Gl, "DataView"), Dl = kl, Nl = C, Ul = $, Kl = Nl(Ul, "Promise"), Hl = Kl, zl = C, Vl = $, ql = zl(Vl, "Set"), Yl = ql, Wl = C, Jl = $, Xl = Wl(Jl, "WeakMap"), Zl = Xl, ne = Dl, oe = ge, se = Hl, ie = Yl, ce = Zl, Dt = I, G = _t, Je = "[object Map]", Ql = "[object Object]", Xe = "[object Promise]", Ze = "[object Set]", Qe = "[object WeakMap]", et = "[object DataView]", eu = G(ne), tu = G(oe), ru = G(se), au = G(ie), nu = G(ce), O = Dt;
(ne && O(new ne(new ArrayBuffer(1))) != et || oe && O(new oe()) != Je || se && O(se.resolve()) != Xe || ie && O(new ie()) != Ze || ce && O(new ce()) != Qe) && (O = function(e) {
  var t = Dt(e), r = t == Ql ? e.constructor : void 0, a = r ? G(r) : "";
  if (a)
    switch (a) {
      case eu:
        return et;
      case tu:
        return Je;
      case ru:
        return Xe;
      case au:
        return Ze;
      case nu:
        return Qe;
    }
  return t;
});
var je = O, ou = Object.prototype, su = ou.hasOwnProperty;
function iu(e) {
  var t = e.length, r = new e.constructor(t);
  return t && typeof e[0] == "string" && su.call(e, "index") && (r.index = e.index, r.input = e.input), r;
}
var cu = iu, lu = $, uu = lu.Uint8Array, fu = uu, tt = fu;
function pu(e) {
  var t = new e.constructor(e.byteLength);
  return new tt(t).set(new tt(e)), t;
}
var Pe = pu, vu = Pe;
function gu(e, t) {
  var r = t ? vu(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.byteLength);
}
var du = gu, hu = /\w*$/;
function bu(e) {
  var t = new e.constructor(e.source, hu.exec(e));
  return t.lastIndex = e.lastIndex, t;
}
var yu = bu, rt = U, at = rt ? rt.prototype : void 0, nt = at ? at.valueOf : void 0;
function $u(e) {
  return nt ? Object(nt.call(e)) : {};
}
var mu = $u, _u = Pe;
function Tu(e, t) {
  var r = t ? _u(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.length);
}
var xu = Tu, Au = Pe, Su = du, Ou = yu, wu = mu, Cu = xu, ju = "[object Boolean]", Pu = "[object Date]", Iu = "[object Map]", Eu = "[object Number]", Mu = "[object RegExp]", Bu = "[object Set]", Fu = "[object String]", Lu = "[object Symbol]", Ru = "[object ArrayBuffer]", Gu = "[object DataView]", ku = "[object Float32Array]", Du = "[object Float64Array]", Nu = "[object Int8Array]", Uu = "[object Int16Array]", Ku = "[object Int32Array]", Hu = "[object Uint8Array]", zu = "[object Uint8ClampedArray]", Vu = "[object Uint16Array]", qu = "[object Uint32Array]";
function Yu(e, t, r) {
  var a = e.constructor;
  switch (t) {
    case Ru:
      return Au(e);
    case ju:
    case Pu:
      return new a(+e);
    case Gu:
      return Su(e, r);
    case ku:
    case Du:
    case Nu:
    case Uu:
    case Ku:
    case Hu:
    case zu:
    case Vu:
    case qu:
      return Cu(e, r);
    case Iu:
      return new a();
    case Eu:
    case Fu:
      return new a(e);
    case Mu:
      return Ou(e);
    case Bu:
      return new a();
    case Lu:
      return wu(e);
  }
}
var Wu = Yu, Ju = E, ot = Object.create, Xu = /* @__PURE__ */ function() {
  function e() {
  }
  return function(t) {
    if (!Ju(t))
      return {};
    if (ot)
      return ot(t);
    e.prototype = t;
    var r = new e();
    return e.prototype = void 0, r;
  };
}(), Zu = Xu, Qu = Zu, ef = Ce, tf = Ae;
function rf(e) {
  return typeof e.constructor == "function" && !tf(e) ? Qu(ef(e)) : {};
}
var af = rf, nf = je, of = w, sf = "[object Map]";
function cf(e) {
  return of(e) && nf(e) == sf;
}
var lf = cf, uf = lf, ff = Te, st = xe, it = st && st.isMap, pf = it ? ff(it) : uf, vf = pf, gf = je, df = w, hf = "[object Set]";
function bf(e) {
  return df(e) && gf(e) == hf;
}
var yf = bf, $f = yf, mf = Te, ct = xe, lt = ct && ct.isSet, _f = lt ? mf(lt) : $f, Tf = _f, xf = vi, Af = di, Sf = be, Of = Bc, wf = Xc, Cf = Zc, jf = el, Pf = pl, If = Al, Ef = El, Mf = kt, Bf = je, Ff = cu, Lf = Wu, Rf = af, Gf = S, kf = Et, Df = vf, Nf = E, Uf = Tf, Kf = Se, Hf = Oe, zf = 1, Vf = 2, qf = 4, Nt = "[object Arguments]", Yf = "[object Array]", Wf = "[object Boolean]", Jf = "[object Date]", Xf = "[object Error]", Ut = "[object Function]", Zf = "[object GeneratorFunction]", Qf = "[object Map]", ep = "[object Number]", Kt = "[object Object]", tp = "[object RegExp]", rp = "[object Set]", ap = "[object String]", np = "[object Symbol]", op = "[object WeakMap]", sp = "[object ArrayBuffer]", ip = "[object DataView]", cp = "[object Float32Array]", lp = "[object Float64Array]", up = "[object Int8Array]", fp = "[object Int16Array]", pp = "[object Int32Array]", vp = "[object Uint8Array]", gp = "[object Uint8ClampedArray]", dp = "[object Uint16Array]", hp = "[object Uint32Array]", f = {};
f[Nt] = f[Yf] = f[sp] = f[ip] = f[Wf] = f[Jf] = f[cp] = f[lp] = f[up] = f[fp] = f[pp] = f[Qf] = f[ep] = f[Kt] = f[tp] = f[rp] = f[ap] = f[np] = f[vp] = f[gp] = f[dp] = f[hp] = !0;
f[Xf] = f[Ut] = f[op] = !1;
function V(e, t, r, a, n, s) {
  var o, i = t & zf, c = t & Vf, l = t & qf;
  if (r && (o = n ? r(e, a, n, s) : r(e)), o !== void 0)
    return o;
  if (!Nf(e))
    return e;
  var u = Gf(e);
  if (u) {
    if (o = Ff(e), !i)
      return jf(e, o);
  } else {
    var g = Bf(e), d = g == Ut || g == Zf;
    if (kf(e))
      return Cf(e, i);
    if (g == Kt || g == Nt || d && !n) {
      if (o = c || d ? {} : Rf(e), !i)
        return c ? If(e, wf(o, e)) : Pf(e, Of(o, e));
    } else {
      if (!f[g])
        return n ? e : {};
      o = Lf(e, g, i);
    }
  }
  s || (s = new xf());
  var x = s.get(e);
  if (x)
    return x;
  s.set(e, o), Uf(e) ? e.forEach(function(h) {
    o.add(V(h, t, r, h, e, s));
  }) : Df(e) && e.forEach(function(h, T) {
    o.set(T, V(h, t, r, T, e, s));
  });
  var m = l ? c ? Mf : Ef : c ? Hf : Kf, _ = u ? void 0 : m(e);
  return Af(_ || e, function(h, T) {
    _ && (T = h, h = e[T]), Sf(o, T, V(h, t, r, T, e, s));
  }), o;
}
var bp = V;
function yp(e) {
  var t = e == null ? 0 : e.length;
  return t ? e[t - 1] : void 0;
}
var $p = yp;
function mp(e, t, r) {
  var a = -1, n = e.length;
  t < 0 && (t = -t > n ? 0 : n + t), r = r > n ? n : r, r < 0 && (r += n), n = t > r ? 0 : r - t >>> 0, t >>>= 0;
  for (var s = Array(n); ++a < n; )
    s[a] = e[a + t];
  return s;
}
var _p = mp, Tp = he, xp = _p;
function Ap(e, t) {
  return t.length < 2 ? e : Tp(e, xp(t, 0, -1));
}
var Sp = Ap, Op = L, wp = $p, Cp = Sp, jp = te;
function Pp(e, t) {
  return t = Op(t, e), e = Cp(e, t), e == null || delete e[jp(wp(t))];
}
var Ip = Pp, Ep = I, Mp = Ce, Bp = w, Fp = "[object Object]", Lp = Function.prototype, Rp = Object.prototype, Ht = Lp.toString, Gp = Rp.hasOwnProperty, kp = Ht.call(Object);
function Dp(e) {
  if (!Bp(e) || Ep(e) != Fp)
    return !1;
  var t = Mp(e);
  if (t === null)
    return !0;
  var r = Gp.call(t, "constructor") && t.constructor;
  return typeof r == "function" && r instanceof r && Ht.call(r) == kp;
}
var Np = Dp, Up = Np;
function Kp(e) {
  return Up(e) ? void 0 : e;
}
var Hp = Kp, zp = St, Vp = bp, qp = Ip, Yp = L, Wp = K, Jp = Hp, Xp = It, Zp = kt, Qp = 1, ev = 2, tv = 4, rv = Xp(function(e, t) {
  var r = {};
  if (e == null)
    return r;
  var a = !1;
  t = zp(t, function(s) {
    return s = Yp(s, e), a || (a = s.length > 1), s;
  }), Wp(e, Zp(e), r), a && (r = Vp(r, Qp | ev | tv, Jp));
  for (var n = t.length; n--; )
    qp(r, t[n]);
  return r;
}), av = rv;
const nv = /* @__PURE__ */ pe(av);
var ov = he;
function sv(e, t, r) {
  var a = e == null ? void 0 : ov(e, t);
  return a === void 0 ? r : a;
}
var iv = sv;
const cv = /* @__PURE__ */ pe(iv), ae = 10, ut = (e = 0) => (t) => `\x1B[${t + e}m`, ft = (e = 0) => (t) => `\x1B[${38 + e};5;${t}m`, pt = (e = 0) => (t, r, a) => `\x1B[${38 + e};2;${t};${r};${a}m`, v = {
  modifier: {
    reset: [0, 0],
    // 21 isn't widely supported and 22 does the same thing
    bold: [1, 22],
    dim: [2, 22],
    italic: [3, 23],
    underline: [4, 24],
    overline: [53, 55],
    inverse: [7, 27],
    hidden: [8, 28],
    strikethrough: [9, 29]
  },
  color: {
    black: [30, 39],
    red: [31, 39],
    green: [32, 39],
    yellow: [33, 39],
    blue: [34, 39],
    magenta: [35, 39],
    cyan: [36, 39],
    white: [37, 39],
    // Bright color
    blackBright: [90, 39],
    gray: [90, 39],
    // Alias of `blackBright`
    grey: [90, 39],
    // Alias of `blackBright`
    redBright: [91, 39],
    greenBright: [92, 39],
    yellowBright: [93, 39],
    blueBright: [94, 39],
    magentaBright: [95, 39],
    cyanBright: [96, 39],
    whiteBright: [97, 39]
  },
  bgColor: {
    bgBlack: [40, 49],
    bgRed: [41, 49],
    bgGreen: [42, 49],
    bgYellow: [43, 49],
    bgBlue: [44, 49],
    bgMagenta: [45, 49],
    bgCyan: [46, 49],
    bgWhite: [47, 49],
    // Bright color
    bgBlackBright: [100, 49],
    bgGray: [100, 49],
    // Alias of `bgBlackBright`
    bgGrey: [100, 49],
    // Alias of `bgBlackBright`
    bgRedBright: [101, 49],
    bgGreenBright: [102, 49],
    bgYellowBright: [103, 49],
    bgBlueBright: [104, 49],
    bgMagentaBright: [105, 49],
    bgCyanBright: [106, 49],
    bgWhiteBright: [107, 49]
  }
};
Object.keys(v.modifier);
const lv = Object.keys(v.color), uv = Object.keys(v.bgColor);
[...lv, ...uv];
function fv() {
  const e = /* @__PURE__ */ new Map();
  for (const [t, r] of Object.entries(v)) {
    for (const [a, n] of Object.entries(r))
      v[a] = {
        open: `\x1B[${n[0]}m`,
        close: `\x1B[${n[1]}m`
      }, r[a] = v[a], e.set(n[0], n[1]);
    Object.defineProperty(v, t, {
      value: r,
      enumerable: !1
    });
  }
  return Object.defineProperty(v, "codes", {
    value: e,
    enumerable: !1
  }), v.color.close = "\x1B[39m", v.bgColor.close = "\x1B[49m", v.color.ansi = ut(), v.color.ansi256 = ft(), v.color.ansi16m = pt(), v.bgColor.ansi = ut(ae), v.bgColor.ansi256 = ft(ae), v.bgColor.ansi16m = pt(ae), Object.defineProperties(v, {
    rgbToAnsi256: {
      value(t, r, a) {
        return t === r && r === a ? t < 8 ? 16 : t > 248 ? 231 : Math.round((t - 8) / 247 * 24) + 232 : 16 + 36 * Math.round(t / 255 * 5) + 6 * Math.round(r / 255 * 5) + Math.round(a / 255 * 5);
      },
      enumerable: !1
    },
    hexToRgb: {
      value(t) {
        const r = /[a-f\d]{6}|[a-f\d]{3}/i.exec(t.toString(16));
        if (!r)
          return [0, 0, 0];
        let [a] = r;
        a.length === 3 && (a = [...a].map((s) => s + s).join(""));
        const n = Number.parseInt(a, 16);
        return [
          /* eslint-disable no-bitwise */
          n >> 16 & 255,
          n >> 8 & 255,
          n & 255
          /* eslint-enable no-bitwise */
        ];
      },
      enumerable: !1
    },
    hexToAnsi256: {
      value: (t) => v.rgbToAnsi256(...v.hexToRgb(t)),
      enumerable: !1
    },
    ansi256ToAnsi: {
      value(t) {
        if (t < 8)
          return 30 + t;
        if (t < 16)
          return 90 + (t - 8);
        let r, a, n;
        if (t >= 232)
          r = ((t - 232) * 10 + 8) / 255, a = r, n = r;
        else {
          t -= 16;
          const i = t % 36;
          r = Math.floor(t / 36) / 5, a = Math.floor(i / 6) / 5, n = i % 6 / 5;
        }
        const s = Math.max(r, a, n) * 2;
        if (s === 0)
          return 30;
        let o = 30 + (Math.round(n) << 2 | Math.round(a) << 1 | Math.round(r));
        return s === 2 && (o += 60), o;
      },
      enumerable: !1
    },
    rgbToAnsi: {
      value: (t, r, a) => v.ansi256ToAnsi(v.rgbToAnsi256(t, r, a)),
      enumerable: !1
    },
    hexToAnsi: {
      value: (t) => v.ansi256ToAnsi(v.hexToAnsi256(t)),
      enumerable: !1
    }
  }), v;
}
const b = fv(), vt = (() => {
  if (!("navigator" in globalThis))
    return 0;
  if (globalThis.navigator.userAgentData) {
    const e = navigator.userAgentData.brands.find(({ brand: t }) => t === "Chromium");
    if (e && e.version > 93)
      return 3;
  }
  return /\b(Chrome|Chromium)\//.test(globalThis.navigator.userAgent) ? 1 : 0;
})(), gt = vt !== 0 && {
  level: vt
}, pv = {
  stdout: gt,
  stderr: gt
};
function vv(e, t, r) {
  let a = e.indexOf(t);
  if (a === -1)
    return e;
  const n = t.length;
  let s = 0, o = "";
  do
    o += e.slice(s, a) + t + r, s = a + n, a = e.indexOf(t, s);
  while (a !== -1);
  return o += e.slice(s), o;
}
function gv(e, t, r, a) {
  let n = 0, s = "";
  do {
    const o = e[a - 1] === "\r";
    s += e.slice(n, o ? a - 1 : a) + t + (o ? `\r
` : `
`) + r, n = a + 1, a = e.indexOf(`
`, n);
  } while (a !== -1);
  return s += e.slice(n), s;
}
const { stdout: dt, stderr: ht } = pv, le = Symbol("GENERATOR"), j = Symbol("STYLER"), N = Symbol("IS_EMPTY"), bt = [
  "ansi",
  "ansi",
  "ansi256",
  "ansi16m"
], P = /* @__PURE__ */ Object.create(null), dv = (e, t = {}) => {
  if (t.level && !(Number.isInteger(t.level) && t.level >= 0 && t.level <= 3))
    throw new Error("The `level` option should be an integer from 0 to 3");
  const r = dt ? dt.level : 0;
  e.level = t.level === void 0 ? r : t.level;
}, hv = (e) => {
  const t = (...r) => r.join(" ");
  return dv(t, e), Object.setPrototypeOf(t, H.prototype), t;
};
function H(e) {
  return hv(e);
}
Object.setPrototypeOf(H.prototype, Function.prototype);
for (const [e, t] of Object.entries(b))
  P[e] = {
    get() {
      const r = J(this, fe(t.open, t.close, this[j]), this[N]);
      return Object.defineProperty(this, e, { value: r }), r;
    }
  };
P.visible = {
  get() {
    const e = J(this, this[j], !0);
    return Object.defineProperty(this, "visible", { value: e }), e;
  }
};
const ue = (e, t, r, ...a) => e === "rgb" ? t === "ansi16m" ? b[r].ansi16m(...a) : t === "ansi256" ? b[r].ansi256(b.rgbToAnsi256(...a)) : b[r].ansi(b.rgbToAnsi(...a)) : e === "hex" ? ue("rgb", t, r, ...b.hexToRgb(...a)) : b[r][e](...a), bv = ["rgb", "hex", "ansi256"];
for (const e of bv) {
  P[e] = {
    get() {
      const { level: r } = this;
      return function(...a) {
        const n = fe(ue(e, bt[r], "color", ...a), b.color.close, this[j]);
        return J(this, n, this[N]);
      };
    }
  };
  const t = "bg" + e[0].toUpperCase() + e.slice(1);
  P[t] = {
    get() {
      const { level: r } = this;
      return function(...a) {
        const n = fe(ue(e, bt[r], "bgColor", ...a), b.bgColor.close, this[j]);
        return J(this, n, this[N]);
      };
    }
  };
}
const yv = Object.defineProperties(() => {
}, {
  ...P,
  level: {
    enumerable: !0,
    get() {
      return this[le].level;
    },
    set(e) {
      this[le].level = e;
    }
  }
}), fe = (e, t, r) => {
  let a, n;
  return r === void 0 ? (a = e, n = t) : (a = r.openAll + e, n = t + r.closeAll), {
    open: e,
    close: t,
    openAll: a,
    closeAll: n,
    parent: r
  };
}, J = (e, t, r) => {
  const a = (...n) => $v(a, n.length === 1 ? "" + n[0] : n.join(" "));
  return Object.setPrototypeOf(a, yv), a[le] = e, a[j] = t, a[N] = r, a;
}, $v = (e, t) => {
  if (e.level <= 0 || !t)
    return e[N] ? "" : t;
  let r = e[j];
  if (r === void 0)
    return t;
  const { openAll: a, closeAll: n } = r;
  if (t.includes("\x1B"))
    for (; r !== void 0; )
      t = vv(t, r.close, r.open), r = r.parent;
  const s = t.indexOf(`
`);
  return s !== -1 && (t = gv(t, n, a, s)), a + t + n;
};
Object.defineProperties(H.prototype, P);
const y = H();
H({ level: ht ? ht.level : 0 });
function mv(e, t) {
  return [
    {
      path: "/advance-api-test",
      method: "get",
      handler: async (r, a) => {
        a.success({
          status: "ok",
          time: (/* @__PURE__ */ new Date()).toISOString(),
          version: "1.0.0",
          message: "Vite Advance API is working!"
        });
      }
    },
    // API文档路由
    {
      path: "/docs",
      method: "get",
      handler: async (r, a) => {
        const n = e.getRoutes().map((o) => ({
          method: o.method,
          path: o.path,
          module: o.moduleName || "未分类",
          description: o.path.endsWith("/docs") ? "API文档接口" : o.path === "/advance-api-test" ? "插件测试接口" : o.description || "用户自定义接口",
          examples: {
            fetch: `fetch('${o.path}', {
  method: '${o.method}',
  headers: {
    'Content-Type': 'application/json'
  }${o.method !== "GET" ? `,
  body: JSON.stringify({
    // your data here
  })` : ""}
})
  .then(response => response.json())
  .then(data => console.log(data));`,
            axios: `axios.${o.method.toLowerCase()}('${o.path}'${o.method !== "GET" ? `, {
  // your data here
}` : ""})
  .then(response => console.log(response.data));`
          }
        })), s = `
<!DOCTYPE html>
<html lang="zh-CN">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>API 接口文档</title>
    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600&display=swap">
    <link rel="stylesheet" href="https://unpkg.com/antd@5/dist/reset.css">
    <style>
        :root {
            --primary: #1677ff;
            --success: #52c41a;
            --warning: #faad14;
            --error: #ff4d4f;
            --border: #f0f0f0;
            --bg: #ffffff;
            --text: rgba(0, 0, 0, 0.88);
            --text-secondary: rgba(0, 0, 0, 0.45);
        }

        * { margin: 0; padding: 0; box-sizing: border-box; }

        body {
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'PingFang SC', 'Hiragino Sans GB',
                'Microsoft YaHei', 'Helvetica Neue', Arial, sans-serif;
            color: var(--text);
            background: #f5f5f5;
        }

        .layout {
            display: flex;
            min-height: 100vh;
        }

        .sidebar {
            width: 280px;
            background: var(--bg);
            border-right: 1px solid var(--border);
            padding: 24px;
            overflow-y: auto;
            position: fixed;
            height: 100vh;
            box-shadow: 0 2px 8px rgba(0, 0, 0, 0.15);
        }

        .main-content {
            flex: 1;
            margin-left: 280px;
            padding: 24px;
            max-width: calc(100% - 280px);
        }

        .header {
            background: var(--bg);
            padding: 24px;
            border-radius: 8px;
            margin-bottom: 24px;
            box-shadow: 0 1px 2px rgba(0, 0, 0, 0.03);
            transition: box-shadow 0.3s;
        }

        .header:hover {
            box-shadow: 0 3px 6px rgba(0, 0, 0, 0.1);
        }

        .title {
            font-size: 20px;
            font-weight: 500;
            margin-bottom: 16px;
            color: var(--text);
        }

        .search {
            width: 100%;
            padding: 8px 12px;
            border: 1px solid var(--border);
            border-radius: 6px;
            margin-bottom: 16px;
            transition: all 0.3s;
        }

        .search:focus {
            outline: none;
            border-color: var(--primary);
            box-shadow: 0 0 0 2px rgba(22, 119, 255, 0.1);
        }

        .api-list {
            display: flex;
            flex-direction: column;
            gap: 16px;
        }

        .api-item {
            background: var(--bg);
            border-radius: 8px;
            border: 1px solid var(--border);
            padding: 20px;
            transition: all 0.3s;
            box-shadow: 0 1px 2px rgba(0, 0, 0, 0.03);
        }

        .api-item:hover {
            transform: translateY(-2px);
            box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);
            border-color: var(--primary);
        }

        .api-header {
            display: flex;
            align-items: center;
            gap: 12px;
            margin-bottom: 16px;
        }

        .method-tag {
            padding: 4px 12px;
            border-radius: 4px;
            font-size: 13px;
            font-weight: 500;
            text-transform: uppercase;
        }

        .get { background: #e6f4ff; color: var(--primary); }
        .post { background: #f6ffed; color: var(--success); }
        .put { background: #fff7e6; color: var(--warning); }
        .delete { background: #fff1f0; color: var(--error); }

        .path {
            font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, monospace;
            font-size: 14px;
            color: var(--text);
            background: #f5f5f5;
            padding: 4px 8px;
            border-radius: 4px;
        }

        .description {
            color: var(--text-secondary);
            font-size: 14px;
            line-height: 1.6;
            margin-bottom: 16px;
            padding: 12px 16px;
            background: #fafafa;
            border-radius: 4px;
        }

        .meta {
            display: flex;
            gap: 16px;
            font-size: 13px;
            color: var(--text-secondary);
        }

        .meta-item {
            display: flex;
            align-items: center;
            gap: 4px;
        }

        .meta-label {
            color: var(--text-secondary);
        }

        .meta-value {
            color: var(--text);
            background: #f5f5f5;
            padding: 2px 8px;
            border-radius: 4px;
        }

        .nav-list {
            list-style: none;
        }

        .nav-item {
            padding: 8px 12px;
            border-radius: 6px;
            cursor: pointer;
            transition: all 0.3s;
            margin-bottom: 8px;
            display: flex;
            align-items: center;
            gap: 8px;
        }

        .nav-item:hover {
            background: #f5f5f5;
        }

        .nav-item.active {
            background: #e6f4ff;
            color: var(--primary);
        }

        @media (max-width: 768px) {
            .layout { flex-direction: column; }
            .sidebar {
                width: 100%;
                height: auto;
                position: static;
                border-right: none;
                border-bottom: 1px solid var(--border);
            }
            .main-content {
                margin-left: 0;
                max-width: 100%;
            }
        }
    </style>
</head>
<body>
    <div class="layout">
        <aside class="sidebar">
            <div class="title">接口导航</div>
            <input type="text" class="search" placeholder="搜索接口..." id="searchInput">
            <ul class="nav-list" id="navList">
                ${n.map(
          (o, i) => `
                    <li class="nav-item" data-index="${i}">
                        <span class="method-tag ${o.method.toLowerCase()}">${o.method}</span>
                        ${o.path}
                    </li>
                `
        ).join("")}
            </ul>
        </aside>

        <main class="main-content">
            <header class="header">
                <div class="title">API 接口文档</div>
                <div class="meta">
                    <div class="meta-item">
                        <span class="meta-label">接口前缀：</span>
                        <span class="meta-value">${t.prefix}</span>
                    </div>
                    <div class="meta-item">
                        <span class="meta-label">接口总数：</span>
                        <span class="meta-value">${n.length}</span>
                    </div>
                </div>
            </header>

            <div class="api-list" id="apiList">
                ${n.map(
          (o) => `
                    <div class="api-item">
                        <div class="api-header">
                            <span class="method-tag ${o.method.toLowerCase()}">${o.method}</span>
                            <span class="path">${o.path}</span>
                        </div>
                        <p class="description">${o.description || "暂无描述"}</p>
                        <div class="meta">
                            <div class="meta-item">
                                <span class="meta-label">所属模块：</span>
                                <span class="meta-value">${o.module}</span>
                            </div>
                        </div>
                    </div>
                `
        ).join("")}
            </div>
        </main>
    </div>

    <script>
        // 搜索功能
        const searchInput = document.getElementById('searchInput');
        const apiItems = document.querySelectorAll('.api-item');
        const navItems = document.querySelectorAll('.nav-item');

        searchInput.addEventListener('input', (e) => {
            const value = e.target.value.toLowerCase();
            
            apiItems.forEach((item, index) => {
                const path = item.querySelector('.path').textContent.toLowerCase();
                const method = item.querySelector('.method-tag').textContent.toLowerCase();
                const description = item.querySelector('.description').textContent.toLowerCase();
                const visible = path.includes(value) || method.includes(value) || description.includes(value);
                
                item.style.display = visible ? 'block' : 'none';
                navItems[index].style.display = visible ? 'block' : 'none';
            });
        });
    <\/script>
</body>
</html>`;
        a.send(s);
      }
    }
  ];
}
function _v(...e) {
  return "/" + e.map((t) => t.replace(/^\/+|\/+$/g, "")).filter(Boolean).join("/");
}
class Tv {
  constructor() {
    this.routes = [];
  }
  add(t, r, a, n) {
    this.routes.push({
      method: t.toUpperCase(),
      path: r,
      moduleName: a,
      description: n
    });
  }
  getRoutes() {
    return this.routes;
  }
  printRoutes() {
    if (this.routes.length === 0) {
      console.log("  😶 暂无注册的接口");
      return;
    }
    this.routes.forEach(({ method: t, path: r, moduleName: a }) => {
      var o;
      const n = {
        GET: y.blue,
        POST: y.green,
        PUT: y.yellow,
        DELETE: y.red
      }, s = ((o = n[t]) == null ? void 0 : o.call(n, t.padEnd(6))) || y.gray(t.padEnd(6));
      console.log(
        `  ${s} ${r}${a ? y.gray(` (${a})`) : ""}`
      );
    });
  }
}
function A(e, t, r = "", a, n, s, o, i, c) {
  const l = _v(r, a), u = e[t.toLowerCase()];
  u && u.bind(e)(l, async (g, d) => {
    const x = d;
    x.success = function(m) {
      return this.json({
        code: 200,
        data: m,
        message: "success"
      });
    }, x.error = function(m, _ = 500) {
      return this.status(_).json({
        code: _,
        message: m,
        data: null
      });
    }, await n(g, x);
  }), o.add(
    t.toUpperCase(),
    `${s}${l}`,
    i,
    c
  );
}
function xv(e, t, r, a) {
  return {
    get: (n, s, o) => A(
      e,
      "get",
      t,
      n,
      s,
      r,
      a,
      "direct模式",
      o
    ),
    post: (n, s, o) => A(
      e,
      "post",
      t,
      n,
      s,
      r,
      a,
      "direct模式",
      o
    ),
    put: (n, s, o) => A(
      e,
      "put",
      t,
      n,
      s,
      r,
      a,
      "direct模式",
      o
    ),
    delete: (n, s, o) => A(
      e,
      "delete",
      t,
      n,
      s,
      r,
      a,
      "direct模式",
      o
    )
  };
}
function jv(e = {}) {
  const t = k.Router(), r = k(), a = e.prefix || "/api", n = new Tv(), s = {
    prefix: a,
    baseUrl: "",
    apiBase: ""
  }, o = mv(n, s), i = {
    router: t,
    app: r,
    express: k,
    uuid: qt,
    _: { pick: zs, omit: nv, get: cv },
    axios: Vt,
    defineRoutes: (c = "", l) => (l.forEach(({ path: u, method: g, handler: d }) => {
      A(
        t,
        g,
        c,
        u,
        d,
        a,
        n,
        "defineRoutes",
        l[0].description
      );
    }), {
      type: "object",
      base: c,
      apis: l
    }),
    getRoutes: () => n.getRoutes(),
    printRoutes: () => n.printRoutes()
  };
  if (o.forEach(({ path: c, method: l, handler: u }) => {
    A(
      t,
      l,
      "",
      c,
      u,
      a,
      n,
      "内置",
      o[0].description
    );
  }), e.setup) {
    const c = e.setup(i);
    (Array.isArray(c) ? c : [c]).forEach((u) => {
      if (!u) return;
      const g = u.base || "", d = `${u.type}模式`;
      switch (u.type) {
        case "object":
          u.apis && u.apis.forEach(({ path: m, method: _, handler: h }) => {
            A(
              t,
              _,
              g,
              m,
              h,
              a,
              n,
              d
            );
          }), u.routes && u.routes.forEach(({ path: m, method: _, handler: h, description: T }) => {
            A(
              t,
              _,
              g,
              m,
              h,
              a,
              n,
              d,
              T
            );
          });
          break;
        case "direct":
          const x = xv(
            t,
            g,
            a,
            n
          );
          u.setup(x);
          break;
      }
    });
  }
  return s.baseUrl = `${s.prefix}`, s.apiBase = `${s.baseUrl}${a}`, r.use(k.json()), r.use(k.urlencoded({ extended: !0 })), r.use(zt(e.cors || { origin: "*" })), r.use(a, t), r.use((c, l, u, g) => {
    console.error(c), u.status(500).json({ error: c.message || "Internal Server Error" });
  }), console.log(`
`), console.log(`🚀 Vite Advance API 插件启动成功！
`), console.log("📡 服务信息"), console.log(`  • 状态：${y.green("运行中")} ✨`), console.log(`  • 前缀：${y.blue(a)} 🔗`), console.log(`
📚 接口列表`), n.printRoutes(), console.log(`
 💡调用示例:`), console.log("http://localhost:5173/api/status"), console.log("💡 温馨提示"), console.log(`  • API 文档：${y.cyan(a + "/docs")} 📖`), console.log("  • Hash 路由模式下请求无需包含 '#' 符号 ⚠️"), e.setup || console.log(
    `  • ${y.yellow("当前为测试模式")}，访问以上地址即可验证插件是否正常工作 🔍`
  ), {
    name: "vite-advance-api",
    configureServer(c) {
      c.middlewares.use(r);
    },
    getServerInfo: () => s
  };
}
export {
  jv as createAdvanceApi
};
